docs/deploying/deploy-production.mdx

```diff
@@ -0,0 +1,198 @@
+toc_max_heading_level: 5
+---
+
+# Optimize Your dApp for Production
+
+When preparing your Scaffold-Stark 2 dApp for production deployment, there are several optimizations you can implement to improve performance and reduce bundle size.
+
+## 1. Verify System Requirements
+
+Before deploying to production, ensure you have the correct versions of all dependencies:
+
+```bash
+# Check versions
+node --version        # >= v18.17
+starknet-devnet --version  # 0.2.3
+scarb --version      # 2.9.2
+snforge --version    # 0.34.0
+```
+
+:::tip Hint
+Using incompatible versions may cause unexpected issues in production. Make sure to align with the compatible versions specified in Scaffold-Stark 2.
+:::tip Hint
+
+## 2. Remove Debug Features
+
+#### 1. Debug Page Removal
+
+The debug page is a development tool that should be removed for production builds. You can remove it by:
+
+Delete the debug page and its components:
+
+```bash
+rm -rf packages/nextjs/app/debug
+```
+
+#### 2. Remove the debug route from your navigation if you've added it manually in any of your components.
+
+:::tip Hint
+If you want to keep the debug features for development while removing them from production, you can use environment-based conditional rendering:
+
+````typescript
+{process.env.NODE_ENV === 'development' && (
+  <Link href="/debug">Debug Contracts</Link>
+)}
+
+:::tip Hint
+
+## 3. Asset optimization
+
+### Image Optimization
+Scaffold-Stark 2 comes with pre-configured image optimization in [`next.config.mjs`](https://github.com/Scaffold-Stark/scaffold-stark-2/blob/main/packages/nextjs/next.config.mjs). The configuration includes secure handling of SVGs and specific remote patterns for Starknet-related images.
+
+#### 1. When adding new image sources, extend the existing `remotePatterns` in `next.config.mjs`:
+
+```javascript
+images: {
+  dangerouslyAllowSVG: true,
+  remotePatterns: [
+    {
+      protocol: "https",
+      hostname: "your-new-domain.com",
+      pathname: "/**",
+    },
+    // existing patterns...
+  ],
+}
+````
+
+#### 2. Use the Next.js Image component with the allowed domains:
+
+```typescript
+import Image from 'next/image';
+
+export function YourComponent() {
+  return (
+    <Image
+      src="https://identicon.starknet.id/your-image"
+      alt="Description"
+      width={400}
+      height={300}
+    />
+  );
+}
+```
+
+:::tip Hint
+Only add domains to `remotePatterns` that you trust and need for your application. For each additional pattern may increases the potential attack surface of the application.
+:::tip Hint
+
+### Additional Build optimizations
+
+Scaffold-Stark 2 includes several build optimizations in `next.config.mjs`:
+
+#### 1. TypeScript and ESLint error handling:
+
+```javascript
+typescript: {
+  ignoreBuildErrors: process.env.NEXT_PUBLIC_IGNORE_BUILD_ERROR === "true",
+},
+eslint: {
+  ignoreDuringBuilds: process.env.NEXT_PUBLIC_IGNORE_BUILD_ERROR === "true",
+},
+```
+
+#### 2. Webpack optimization for compatibility:
```

Comment:
I think need more information here, considering many beginners would want to use the framework